OData/src/System.Web.OData/OData/Batch/DefaultODataBatchHandler.cs

```diff
@@ -42,7 +44,12 @@ public DefaultODataBatchHandler(HttpServer httpServer)
             IList<ODataBatchRequestItem> subRequests = await ParseBatchRequestsAsync(request, cancellationToken);
 
             string preferHeader = RequestPreferenceHelpers.GetRequestPreferHeader(request);
-            if (preferHeader != null && preferHeader.Contains(PreferenceContinueOnError))
+            var configuration = request.GetConfiguration();
+            if (configuration == null)
+            {
+                throw Error.InvalidOperation(SRResources.RequestMustContainConfiguration);
```

Comment:
Can we not throw this exception and let it do the following if configuration == null